app/src/main/java/io/apicurio/registry/rules/compatibility/GenericCompatibilityDifference.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.rules.compatibility;
+
+import io.apicurio.registry.rest.beans.RuleViolationCause;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+public class GenericCompatibilityDifference implements CompatibilityDifference {
+
+    private final String message;
+    
+    /**
+     * Constructor.
+     * @param e
+     */
+    public GenericCompatibilityDifference(Exception e) {
+        this.message = e.getMessage();
+    }
+
+    /**
+     * Constructor.
+     * @param message
+     */
+    public GenericCompatibilityDifference(String message) {
+        this.message = message;
+    }
+    
+    /**
+     * @see io.apicurio.registry.rules.compatibility.CompatibilityDifference#asRuleViolationCause()
+     */
+    @Override
+    public RuleViolationCause asRuleViolationCause() {
+        RuleViolationCause cause = new RuleViolationCause();
+        cause.setDescription(message);
+        cause.setContext("/");
```

Comment:
What does such context mean?

Reply:
The context is intended to mean where in the document the error/issue exists.  So basically a "path" into the document to help identify the location of the problem.  For most of our compatibility support we don't have that information.  For example, the Avro libraries don't provide a lot of feedback when doing this compatibility check - they give you a "yes" or "no".  So in these cases we just have a high level error message and  use "/" to indicate the root or entire document.